### PR TITLE
Memory leak

### PR DESCRIPTION
**Pull request by RobPruzan**
Fixes memory leaked caused by frequent accumulation of re-renders in a large array

- aggregates renders together in scheduled/active outlines immediately, instead of doing it when painting.
  - this required a large rewrite of painting logic
- re-use objects when possible
- remove mergeOutline merge limit

## Before

https://github.com/user-attachments/assets/748a228e-9ea3-45fc-b93f-2991310180c9

https://github.com/user-attachments/assets/f001647d-e60e-4352-baeb-daa32003df6c

### CODE REVIEW

File: packages/scan/src/core/index.ts [hunk 1]
```diff
@@ -340,7 +346,7 @@ export const reportRender = (fiber: Fiber, renders: Array<Render>) => {
   const { selfTime } = getTimings(fiber.type);
   const displayName = getDisplayName(fiber.type);
 
-  Store.lastReportTime.value = performance.now();
+  Store.lastReportTime.value = Date.now();
```
Comment (RobPruzan) [inline packages/scan/src/core/index.ts:349]: performance.now() is slow and we don't need the precision

File: packages/scan/src/core/instrumentation.ts [hunk 2]
```diff
@@ -75,23 +75,29 @@ export const isElementInViewport = (
...
   type: 'props' | 'context' | 'state';
   name: string;
   prevValue: unknown;
   nextValue: unknown;
   unstable: boolean;
 }
 
+// this must grow O(1) w.r.t to renders
+export interface AggregatedChange {
+  type: Set<'props' | 'context' | 'state'>;
```
Comment (lxsmnsyc) [inline packages/scan/src/core/instrumentation.ts:88]: hmmm this in the form of bit flags might be possible here

File: packages/scan/src/core/utils.ts [hunk 3]
```diff
@@ -53,6 +85,7 @@ export const getLabelText = (renders: Array<Render>) => {
   const parts: Array<string> = [];
   let cumulativeTime = 0;
   for (const count of sortedCounts) {
+    // biome-ignore lint/style/noNonNullAssertion: <explanation>
```
Comment (lxsmnsyc) [inline packages/scan/src/core/utils.ts:88]: weird, the linter should've been disabled in main

Comment (RobPruzan) [reply packages/scan/src/core/utils.ts:88]: oh that's probably old before i rebased

Comment (RobPruzan) [reply packages/scan/src/core/utils.ts]: yeah it is, removed them

File: packages/scan/src/core/monitor/debug.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
+(function () {
+  function getInteractionType(eventName) {
+    if (['pointerdown', 'pointerup', 'click'].includes(eventName)) {
```
Comment (lxsmnsyc) [inline packages/scan/src/core/monitor/debug.js:3]: switch might be better in this whole function

File: packages/scan/src/core/web/utils/helpers.ts [hunk 5]
```diff
@@ -63,7 +60,7 @@ export const debounce = <T extends (...args: Array<any>) => any>(
       timeoutId = window.setTimeout(() => {
         isLeadingInvoked = false;
         timeoutId = undefined;
-        fn(...(lastArgs!));
+        fn(...lastArgs!);
```
Comment (lxsmnsyc) [inline packages/scan/src/core/web/utils/helpers.ts:63]: `fn.apply(null, lastArgs)`, but the linter might bork

File: packages/scan/src/core/monitor/debug.js (hunk 4 above)
Comment (RobPruzan) [reply packages/scan/src/core/monitor/debug.js:3]: oops my b that was just to debug something i didn't mean to commit it

File: packages/scan/src/core/web/utils/helpers.ts [hunk 6]
```diff
@@ -129,6 +112,9 @@ export const saveLocalStorage = <T>(storageKey: string, state: T): | void => {
   }
 };
 
-export const toggleMultipleClasses = (element: HTMLElement, ...classes: Array<string>) => {
-  classes.forEach(cls => element.classList.toggle(cls));
+export const toggleMultipleClasses = (
+  element: HTMLElement,
+  ...classes: Array<string>
+) => {
+  classes.forEach((cls) => element.classList.toggle(cls));
```
Comment (lxsmnsyc) [inline packages/scan/src/core/web/utils/helpers.ts:119]: `for..of` is fine. Honestly let's just keep from using `forEach` on all iterables

Comment (RobPruzan) [reply packages/scan/src/core/web/utils/helpers.ts:119]: just formatting isn't my code

File: packages/scan/src/core/web/utils/outline.ts [hunk 7]
```diff
@@ -391,178 +301,281 @@ export const fadeOutOutline = (
...
-) => {
-  const newPreviousOutlines = new Map<string, PendingOutline>();
+const activateOutlines = async () => {
+  const domNodes: Array<HTMLElement> = [];
+  const scheduledOutlines = ReactScanInternals.scheduledOutlines;
+  const activeOutlines = ReactScanInternals.activeOutlines;
+  const activeFibers = new Map<Fiber, AggregatedRender>();
+
+  for (const activeOutline of ReactScanInternals.activeOutlines.values()) {
+    activeOutline.groupedAggregatedRender?.forEach(
```
Comment (lxsmnsyc) [inline packages/scan/src/core/web/utils/outline.ts:351]: `for..of` here

File: packages/scan/src/core/web/utils/helpers.ts (hunk 5 above)
Comment (RobPruzan) [reply packages/scan/src/core/web/utils/helpers.ts:63]: same thing as other comment, just formatting not my code

File: packages/scan/src/core/web/utils/helpers.ts (hunk 6 above)
Comment (lxsmnsyc) [reply packages/scan/src/core/web/utils/helpers.ts:119]: oof then

**Pull request merged**